app/assets/javascripts/searchform.js

```diff
@@ -10,3 +10,21 @@ $(function() {
     window.location = "/search/" + encoded_query;
   });
 });
+
+$(function () {
+  $('body').on('keydown', '#searchform_input', function (e) {
+    if (e.which === 32 && this.value === '') {
+      return false;
+    }
+  });
+});
+
+function moreChar(textbox) {
+  var x = textbox.value;
+  if(x.length<=2 && x.length>0){
```

Comment:
```suggestion
  if(x.length <= 2 && x.length > 0){
```
Let's clean it a bit!